alphastats/DataSet_Preprocess.py

```diff
@@ -9,16 +10,55 @@
 import streamlit as st
 
 from sklearn.experimental import enable_iterative_imputer  # noqa
-from tqdm import tqdm
 
 from alphastats.utils import ignore_warning
 
 
-class Preprocess:
+class PreprocessInterface(ABC):
+    @abstractmethod
+    def preprocess(
+        self,
+        log2_transform: bool,
+        remove_contaminations: bool,
+        subset: bool,
+        data_completeness: float,
+        normalization: str,
+        imputation: str,
+        remove_samples: list,
+        **kwargs,
+    ):
+        pass
+
+    @abstractmethod
+    def batch_correction(self, batch: str) -> None:
+        pass
+
+
+class Preprocess(PreprocessInterface):
     imputation_methods = ["mean", "median", "knn", "randomforest"]
     normalization_methods = ["vst", "zscore", "quantile"]
 
-    def _remove_sampels(self, sample_list: list):
+    def __init__(
+        self,
+        filter_columns,
+        rawinput,
+        index_column,
+        sample,
+        metadata,
+        preprocessing_info,
+        mat,
+    ):
+        self.filter_columns = filter_columns
```

Comment:
Are these required for preprocessing? Otherwise I would not include them here.

Reply:
they are, in the `_filter` method